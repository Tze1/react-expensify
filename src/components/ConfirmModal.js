import React from 'react';
import { Modal, Button } from 'react-bootstrap';

/* Usage:
* import ConfirmModel from '<path-from-importer-to-ConfirmModal>';
* <Modal
*   show={this.state.showConfirmModal}
*   title='<your-optional-title>'
*   content='<your-required-content>'
*   btnLabelCancel='<your-optional-cancel-btn-label>'
*   btnLabelOk='<your-optional-ok-btn-label>'
*   handleCancel={this.handleCancel}
*   handleOk={this.handleOk}
* />
*/

const ConfirmModal = (props) => (
    <Modal show={ props.show } className="confirmmodal component">
      <Modal.Header className="modal-header">
        <Modal.Title className="modal-title">{ props.title ? props.title : 'Confirm'}</Modal.Title>
      </Modal.Header>

      <Modal.Body className="modal-body">
        <p className="modal-message">{ props.content }</p>
      </Modal.Body>

      <Modal.Footer className="modal-footer">
        <Button
          onClick={ props.handleCancel }
          className="modal-cancel button exit"
        >
          { props.btnLabelCancel ? props.btnLabelCancel : 'Cancel' }
        </Button>
        <Button
          onClick={ props.handleOk }
          className="modal-ok button danger"
        >
          { props.btnLabelOk ? props.btnLabelOk : 'OK' }
        </Button>
      </Modal.Footer>
    </Modal>
);

export default ConfirmModal;
