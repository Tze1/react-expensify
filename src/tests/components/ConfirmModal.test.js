import React from 'react';
import { shallow } from 'enzyme';
import ConfirmModal from '../../components/ConfirmModal';

const handleOkStub = jest.fn();
const handleCancelStub = jest.fn();
const requiredProps = {
  show: true,
  content: 'Are you sure you want to do this?',
};
const optionalProps = {
  title: 'test',
  btnLabelCancel: 'no',
  btnLabelOk: 'yes',
};

const wrapper = shallow(<ConfirmModal
  handleOk={ handleOkStub }
  handleCancel={ handleCancelStub }
/>);
wrapper.setProps(requiredProps);

test('Should render properly with default prop values', () => {
  expect(wrapper).toMatchSnapshot();
});

test('Should render properly with optional prop values', () => {
  wrapper.setProps(optionalProps);

  expect(wrapper).toMatchSnapshot();
});

test('Should call cancel method on cancel-button click', () => {
  handleOkStub.mockClear();
  handleCancelStub.mockClear();
  wrapper.find('button.modal-cancel').simulate('click');

  expect(handleOkStub).not.toHaveBeenCalled();
  expect(handleCancelStub).toHaveBeenCalled();
});

test('Should call ok method on ok-button click', () => {
  handleOkStub.mockClear();
  handleCancelStub.mockClear();
  wrapper.find('button.modal-ok').simulate('click');

  expect(handleCancelStub).not.toHaveBeenCalled();
  expect(handleOkStub).toHaveBeenCalled();
});
