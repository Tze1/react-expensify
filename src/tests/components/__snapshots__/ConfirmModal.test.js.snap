// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render properly with default prop values 1`] = `
<Bootstrap(Modal)
  className="confirmmodal component"
  show={true}
>
  <ModalHeader
    className="modal-header"
    closeButton={false}
    closeLabel="Close"
  >
    <ModalTitle
      className="modal-title"
    >
      Confirm
    </ModalTitle>
  </ModalHeader>
  <ModalBody
    className="modal-body"
  >
    <p
      className="modal-message"
    >
      Are you sure you want to do this?
    </p>
  </ModalBody>
  <ModalFooter
    className="modal-footer"
  >
    <Button
      active={false}
      className="modal-cancel button exit"
      disabled={false}
      onClick={[MockFunction]}
      type="button"
      variant="primary"
    >
      Cancel
    </Button>
    <Button
      active={false}
      className="modal-ok button danger"
      disabled={false}
      onClick={[MockFunction]}
      type="button"
      variant="primary"
    >
      OK
    </Button>
  </ModalFooter>
</Bootstrap(Modal)>
`;

exports[`Should render properly with optional prop values 1`] = `
<Bootstrap(Modal)
  className="confirmmodal component"
  show={true}
>
  <ModalHeader
    className="modal-header"
    closeButton={false}
    closeLabel="Close"
  >
    <ModalTitle
      className="modal-title"
    >
      test
    </ModalTitle>
  </ModalHeader>
  <ModalBody
    className="modal-body"
  >
    <p
      className="modal-message"
    >
      Are you sure you want to do this?
    </p>
  </ModalBody>
  <ModalFooter
    className="modal-footer"
  >
    <Button
      active={false}
      className="modal-cancel button exit"
      disabled={false}
      onClick={[MockFunction]}
      type="button"
      variant="primary"
    >
      no
    </Button>
    <Button
      active={false}
      className="modal-ok button danger"
      disabled={false}
      onClick={[MockFunction]}
      type="button"
      variant="primary"
    >
      yes
    </Button>
  </ModalFooter>
</Bootstrap(Modal)>
`;
